packages/ui/src/modal/buy/BuyModal.tsx

```diff
@@ -93,10 +95,24 @@ export function BuyModal({
     openState ? openState[0] : false,
     openState
   )
-  const { chain: activeChain } = useNetwork()
+
+  const { chains, chain: activeChain } = useNetwork()
+  const { switchNetworkAsync } = useSwitchNetwork()
+
+  const selectedChain = chainId
+    ? chains.find((chain) => chain.id === chainId) || activeChain
+    : activeChain
+
+  const handleBuy = async (callback: () => void): Promise<void> => {
+    if (selectedChain?.id !== activeChain?.id) {
+      await switchNetworkAsync?.(selectedChain?.id)
```

Comment:
Update this to make sure the new chain is the right one, otherwise just ignore it.